tests/unit/test_project_structure.py

```diff
@@ -0,0 +1,153 @@
+                for col in ret.columns
+            ]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    @patch('mindsdb.integrations.handlers.postgres_handler.Handler')
+    def test_flow(self, data_handler):
+        # set up
+
+        df = pd.DataFrame([
+            {'a': 1, 'b': dt.datetime(2020, 1, 1)},
+            {'a': 2, 'b': dt.datetime(2020, 1, 2)},
+            {'a': 1, 'b': dt.datetime(2020, 1, 3)},
+        ])
+        self.set_handler(data_handler, name='pg', tables={'tasks': df})
+
+        # ----------------
+
+        # create folder
+        self.run_sql('create database proj')
+
+        # -- create model --
+        self.run_sql(
+            '''
+                CREATE PREDICTOR proj.task_model
+                from pg (select * from tasks)
+                PREDICT a
+                using engine='dummy_ml'
+            '''
+        )
+        self.wait_predictor('proj', 'task_model')
+
+        assert data_handler().native_query.call_args[0][0] == 'select * from tasks'
+
+        # use model
+        ret = self.run_sql('''
+             SELECT m.*
+               FROM pg.tasks as t
+               JOIN proj.task_model as m
+        ''')
+
+        assert len(ret) == 3
+        assert ret.predicted[0] == 42
+
+        # -- retrain predictor with tag --
+        data_handler.reset_mock()
+        self.run_sql(
+            '''
+                retrain proj.task_model
+                from pg (select * from tasks where a=2)
+                PREDICT b
+                using tag = 'new'
+            '''
+        )
+        self.wait_predictor('proj', 'task_model', {'tag': 'new'})
+
+        # get current model
+        ret = self.run_sql('select * from proj.models')
+
+        # check target
+        assert ret['PREDICT'][0] == 'b'
+
+        # check label
+        assert ret['TAG'][0] == 'new'
+
+        # check integration sql
+        assert data_handler().native_query.call_args[0][0] == 'select * from tasks where a=2'
+
+        # use model
+        ret = self.run_sql('''
+             SELECT m.*
+               FROM pg.tasks as t
+               JOIN proj.task_model as m
+        ''')
+        assert ret.predicted[0] == 42
+
+        # -- retrain again with active=0 --
+        data_handler.reset_mock()
+        self.run_sql(
+            '''
+                retrain proj.task_model
+                from pg (select * from tasks where a=2)
+                PREDICT a
+                using tag='new2', active=0
+            '''
+        )
+        self.wait_predictor('proj', 'task_model', {'tag': 'new2'})
+
+        ret = self.run_sql('select * from proj.models')
+
+        # check target is from previous retrain
+        assert ret['PREDICT'][0] == 'b'
+
+        # list of versions
+        ret = self.run_sql('select * from proj.models_versions')
+        # we have all tags in versions
+        assert set(ret['TAG']) == {None, 'new', 'new2'}
+
+        # TODO:
```

Comment:
create an issue for the whole technical debt which you mention in unittests as TODO and link it to the PR description and also to the description of the appropriate Github issue which the PR solves

Reply:
it is not debt, it is not is implemented functions and this part of test scenario will be written in next PR.